lib/storage/src/content_manager/raft_state.rs

```diff
@@ -109,6 +111,19 @@ impl Persistent {
         self.save()
     }
 
+    pub fn set_peer_address_by_id(
+        &mut self,
+        peer_address_by_id: PeerAddressById,
+    ) -> Result<(), StorageError> {
+        self.peer_address_by_id = PeerAddressByIdWrapper(peer_address_by_id);
+        self.save()
+    }
+
+    pub fn insert_peer(&mut self, peer_id: PeerId, address: Uri) -> Result<(), StorageError> {
+        self.peer_address_by_id.0.insert(peer_id, address);
```

Comment:
Random idea: what about having a log in the case where the peer was already present but with a different address?
Or is there a real invariant to assert here?